src/symbolic/symbolic_stencils.jl

```diff
@@ -0,0 +1,94 @@
+```julia
+stencil = CartesianIndex.([(0,1), (0,1), (1,1)])
+est = SpatiotemporalPermutation(stencil)
+```
+Here the stencil creates a 2x2 square extending to the bottom and right of the pixel.
+Notice that no offset (meaning the pixel itself) is always included automatically.
+The length of the stencil decides the order of the permutation entropy, and the ordering
+within the stencil dictates the order that pixels are compared with.
+The pixel without any offset is always first in the order.
+
+After having defined `est`, one calculates the permutation entropy of ordinal patterns
+by calling [`genentropy`](@ref) with `est`, and with the array data.
+To apply this to timeseries of spatial data, simply loop over the call, e.g.:
+```julia
+data = [rand(50, 50) for _ in 1:50]
+entropies = [genentropy(x, est) for x in data]
+```
+
+The argument `periodic` decides whether the stencil should wrap around at the end of the
+array. If `periodic = false`, pixels whose stencil exceeds the array bounds are skipped.
+
+[^Ribeiro2012]:
+    Ribeiro et al. (2012). Complexity-entropy causality plane as a complexity measure
+    for two-dimensional patterns. https://doi.org/10.1371/journal.pone.0040689
+
+[^Schlemmer2018]:
+    Schlemmer et al. (2012). Spatiotemporal Permutation Entropy as a Measure for
+    Complexity of Cardiac Arrhythmia. https://doi.org/10.3389/fphy.2018.00039
+"""
+struct SpatiotemporalPermutation{D,P} <: ProbabilitiesEstimator
+    stencil::Vector{CartesianIndex{D}}
+    viewer::Vector{CartesianIndex{D}}
+end
+function SpatiotemporalPermutation(
+        stencil::Vector{CartesianIndex{D}}, p::Bool = true
+    ) where {D}
+    # Ensure that no offset is part of the stencil
+    stencil = pushfirst!(copy(stencil), CartesianIndex{D}(zeros(Int, D)...))
+    SpatiotemporalPermutation{D, p}(stencil, copy(stencil))
+end
+
+# This source code is a modification of the code of Agents.jl that finds neighbors
+# in grid-like spaces. It's the code of `nearby_positions` in `grid_general.jl`.
+function pixels_in_stencil(array, pixel, spatperm::SpatiotemporalPermutation{D,false}) where {D}
+    # TODO: Not clear yet how out of bounds is handled in this case.
+    @inbounds for i in eachindex(spatperm.stencil)
+        statperm.viewer[i] = spatperm.stencil[i] + pixel
+    end
+    return statperm.viewer
+    # space_size = size(array)
+    # positions_iterator = (n + pixel for n in spatperm.stencil)
+    # return Base.Iterators.filter(
+    #     pos -> checkbounds(Bool, array, pos...), positions_iterator
+    # )
+end
+
+function pixels_in_stencil(array, pixel, spatperm::SpatiotemporalPermutation{D,true}) where {D}
+    space_size = size(array)
+    @inbounds for i in eachindex(spatperm.stencil)
+        spatperm.viewer[i] = CartesianIndex{D}(mod1.(Tuple(spatperm.stencil[i] + pixel), space_size))
+    end
+    return spatperm.viewer
+end
+
+function Entropies.probabilities(x, est::SpatiotemporalPermutation) where {m, T}
+    s = zeros(Int, length(x))
```

Comment:
@kahaaga Is this correct? Should `s` be this length? I guess for the periodic it is correct, as each pixel gives one pattetrn, but for the nnon periodic I should add a second clause.